README.md

```diff
@@ -8,22 +8,845 @@
+- The last random 11 bytes (i.e. `1212121212121212121212`) allow for $2^{88}$ bits of entropy for mining a salt.
+
+The full logic is implemented in the `internal` [`_guard`](./src/CreateX.sol#L879-L917) function:
+
+```solidity
+function _guard(bytes32 salt) internal view returns (bytes32 guardedSalt) {
+  (
+    SenderBytes senderBytes,
+    RedeployProtectionFlag redeployProtectionFlag
+  ) = _parseSalt({ salt: salt });
+
+  if (
+    senderBytes == SenderBytes.MsgSender &&
+    redeployProtectionFlag == RedeployProtectionFlag.True
+  ) {
+    // Configures a permissioned deploy protection as well as a cross-chain redeploy protection.
+    guardedSalt = keccak256(abi.encode(msg.sender, block.chainid, salt));
+  } else if (
+    senderBytes == SenderBytes.MsgSender &&
+    redeployProtectionFlag == RedeployProtectionFlag.False
+  ) {
+    // Configures solely a permissioned deploy protection.
+    guardedSalt = _efficientHash({
+      a: bytes32(uint256(uint160(msg.sender))),
+      b: salt
+    });
+  } else if (senderBytes == SenderBytes.MsgSender) {
+    // Reverts if the 21st byte is greater than `0x01` in order to enforce developer explicitness.
+    revert InvalidSalt({ emitter: _SELF });
+  } else if (
+    senderBytes == SenderBytes.ZeroAddress &&
+    redeployProtectionFlag == RedeployProtectionFlag.True
+  ) {
+    // Configures solely a cross-chain redeploy protection. In order to prevent a pseudo-randomly
+    // generated cross-chain redeploy protection, we enforce the zero address check for the first 20 bytes.
+    guardedSalt = _efficientHash({ a: bytes32(block.chainid), b: salt });
+  } else if (
+    senderBytes == SenderBytes.ZeroAddress &&
+    redeployProtectionFlag == RedeployProtectionFlag.Unspecified
+  ) {
+    // Reverts if the 21st byte is greater than `0x01` in order to enforce developer explicitness.
+    revert InvalidSalt({ emitter: _SELF });
+  } else {
+    // In all other cases, the salt value `salt` is not modified.
+    guardedSalt = salt;
+  }
+}
+```
+
+Please note that when you configure a permissioned deploy protection, you **must** specify whether you want a cross-chain redeploy protection or not (i.e. the 21st byte equals `0x00`). The underlying reason for this logic is to enforce developer explicitness. If you don't configure a cross-chain redeploy protection, i.e. the 21st byte is greater than `0x01`, the function reverts. Furthermore, you can configure solely a cross-chain redeploy protection by setting the first 20 bytes equal to the zero address `0x0000000000000000000000000000000000000000`. The rationale behind this logic is to prevent a pseudo-randomly generated cross-chain redeploy protection. Also in this case, if you don't specify a cross-chain redeploy protection, i.e. the 21st byte is greater than `0x01`, the function reverts. The underlying reason for this logic is as well to enforce developer explicitness.
```

Comment:
Nit on phrasing, I think this might be a more clear
```suggestion
Please note that when you configure a permissioned deploy protection, you **must** specify whether you want cross-chain redeploy protection (i.e. 21st byte equals `0x01`) or not (i.e. the 21st byte equals `0x00`). The underlying reason for this logic is to enforce developer explicitness. If you don't specify a cross-chain redeploy protection decision (i.e. the 21st byte is greater than `0x01`) the function reverts.

Furthermore, you can configure _only_ cross-chain redeploy protection by setting the first 20 bytes equal to the zero address `0x0000000000000000000000000000000000000000`. The rationale behind this logic is to prevent a pseudo-randomly generated 32 byte salt from inadvertently activating cross-chain redeploy protection. Also in this case, if you don't specify a cross-chain redeploy protection, i.e. the 21st byte is greater than `0x01`, the function reverts. The underlying reason for this logic is as well to enforce developer explicitness.
```